frontend/app/src/next/pages/authenticated/dashboard/workers/managed/components/metrics-tab.tsx

```diff
@@ -0,0 +1,246 @@
+  }, [createdAfter, finishedBefore, setMetricsQuery]);
+
+  // Auto-update time range
+  useEffect(() => {
+    const interval = setInterval(() => {
+      if (customTimeRange) {
+        return;
+      }
+      setCreatedAfter(getCreatedAfterFromTimeRange(defaultTimeRange));
+    }, 60 * 1000);
+
+    return () => clearInterval(interval);
+  }, [defaultTimeRange, customTimeRange]);
+
+  // Handle time range changes
+  useEffect(() => {
+    if (customTimeRange && customTimeRange.length === 2) {
+      setCreatedAfter(customTimeRange[0]);
+      setFinishedBefore(customTimeRange[1]);
+    } else if (defaultTimeRange) {
+      setCreatedAfter(getCreatedAfterFromTimeRange(defaultTimeRange));
+      setFinishedBefore(undefined);
+    }
+  }, [defaultTimeRange, customTimeRange]);
+
+  if (
+    !managedWorker ||
+    metrics?.cpu?.isLoading ||
+    metrics?.memory?.isLoading ||
+    metrics?.disk?.isLoading ||
+    !metrics?.cpu?.data ||
+    !metrics?.memory?.data ||
+    !metrics?.disk?.data
+  ) {
+    return <Spinner />;
+  }
+
+  return (
+    <div className="flex flex-col gap-4">
+      <div className="flex flex-row justify-between items-center">
+        <h3 className="text-xl font-bold leading-tight text-foreground">
+          Metrics
+        </h3>
+        <div className="flex flex-row justify-end items-center gap-2">
+          {customTimeRange
+            ? [
+                <Button
+                  key="clear"
+                  onClick={() => {
+                    setCustomTimeRange(undefined);
+                  }}
+                  variant="outline"
+                  size="sm"
+                  className="text-xs h-9 py-2"
+                >
+                  <XCircleIcon className="h-[18px] w-[18px] mr-2" />
+                  Clear
+                </Button>,
+                <DateTimePicker
+                  key="after"
+                  label="After"
+                  date={createdAfter ? new Date(createdAfter) : undefined}
+                  setDate={(date) => {
+                    setCreatedAfter(date?.toISOString() || '');
+                  }}
+                />,
+                <DateTimePicker
+                  key="before"
+                  label="Before"
+                  date={finishedBefore ? new Date(finishedBefore) : undefined}
+                  setDate={(date) => {
+                    setFinishedBefore(date?.toISOString());
+                  }}
+                />,
+              ]
+            : null}
+          <Select
+            value={customTimeRange ? 'custom' : defaultTimeRange}
+            onValueChange={(value) => {
+              if (value !== 'custom') {
+                setDefaultTimeRange(value);
+                setCustomTimeRange(undefined);
+              } else {
+                setCustomTimeRange([
+                  getCreatedAfterFromTimeRange(defaultTimeRange),
+                  new Date().toISOString(),
+                ]);
+              }
+            }}
+          >
```

Comment:
I think ideally we'd be able to use the existing `TimeFilter` and `TimeFilterGroup` for this